25-26 InitialSetup

samarthmaheshmenon opened the pull request:
Created 25-26 Manager/Base classes as well as some additional files: ManualManager, AutoCodeTest, Log Manager (Build is successful but has not been tested)

Before issuing a pull request, please see the contributing page.

samarthmaheshmenon committed:
Created 25-26 Manager/Base classes as well as some additional files; ManualManager, AutoCodeTest, Log Manager (Build is successful but has not been tested)

akhanna02 commented:
This code shouldn't be merged as is. There are lots of part of the code from last year that don't apply this year.

samarthmaheshmenon commented:
> This code shouldn't be merged as is. There are lots of part of the code from last year that don't apply this year.

This was not merged as is. You can see it is more structured and object oriented. The classes are now defined even for manual unlike last year. If you are talking about viper slide, arm manager and such, I have left those there intentionally so in a similar way we can create this years component managers. I can change it now that we know all our components though unless someone else wants to do it.

akhanna02 commented:
I generally recommend starting a clean slate. I would only take a robothardware and the manual class into this branch and get that merged. You don't want to bring the baggage from the other stale code.

Once that is good then merge into the main as needed.

samarthmaheshmenon commented:
RoboHardware and Manual Op were not copied just like that. It has been written in a more OO way and named accordingly as you can see. There are other classes that were added as well. I had gone over the changes with the mentor and team over the summer when I cleaned it up as part of Mr.Abhijit's assignment. We didn't know about this season's task then, which is why I left the last season's components in case we can reuse it for this season. That being said, I have the PR with the components for this season and can go over it with the team and i have added you all as reviewers of that. Please provide feedback there. Once I get an approval,  I can merge that into the master branch. What's remaining is testing the changes I have made.
